core/src/banking_stage/transaction_scheduler/scheduler_metrics.rs

```diff
@@ -332,3 +334,71 @@ impl SchedulerTimingMetricsInner {
         self.receive_completed_time_us = 0;
     }
 }
+
+#[derive(Default)]
+pub struct SchedulerLeaderDetectionMetrics {
+    inner: Option<SchedulerLeaderDetectionMetricsInner>,
+}
+
+pub struct SchedulerLeaderDetectionMetricsInner {
+    slot: Slot,
+    bank_detected_time: Instant,
+    bank_detected_delay_us: u64,
+}
+
+impl SchedulerLeaderDetectionMetrics {
+    pub fn update_and_maybe_report(&mut self, bank_start: Option<&BankStart>) {
+        match (&self.inner, bank_start) {
+            (None, Some(bank_start)) => self.initialize_inner(bank_start),
+            (Some(_inner), None) => self.report_and_reset(),
+            (Some(inner), Some(bank_start)) if inner.slot != bank_start.working_bank.slot() => {
+                self.report_and_reset();
+                self.initialize_inner(bank_start);
+            }
+            _ => {}
+        }
+    }
+
+    fn initialize_inner(&mut self, bank_start: &BankStart) {
+        let bank_detected_time = Instant::now();
+        let bank_detected_delay_us = bank_detected_time
+            .duration_since(*bank_start.bank_creation_time)
+            .as_micros()
+            .try_into()
+            .unwrap_or(u64::MAX);
+        self.inner = Some(SchedulerLeaderDetectionMetricsInner {
+            slot: bank_start.working_bank.slot(),
+            bank_detected_time,
+            bank_detected_delay_us,
+        });
+    }
+
+    fn report_and_reset(&mut self) {
+        let SchedulerLeaderDetectionMetricsInner {
+            slot,
+            bank_detected_time,
+            bank_detected_delay_us,
+        } = self.inner.take().expect("inner must be present");
+
+        let bank_detected_to_slot_end_detected_us = bank_detected_time
+            .elapsed()
+            .as_micros()
+            .try_into()
+            .unwrap_or(u64::MAX);
+        datapoint_info!(
+            "banking_stage_scheduler_leader_detection",
+            ("slot", slot, i64),
+            ("bank_detected_delay_us", bank_detected_delay_us, i64),
+            (
+                "bank_detected_to_slot_end_detected_us",
+                bank_detected_to_slot_end_detected_us,
+                i64
+            ),
+            (
+                "bank_creation_to_slot_end_detected_us",
+                bank_detected_to_slot_end_detected_us + bank_detected_delay_us,
+                i64
+            ),
```

Comment:
nit: i dunno enough context. but `bank_creation_to_slot_end_detected_us` can be computed at influxdb side easily with the preceding fields. yet is this needed to be reported redundantly for some ops convenience maybe?

i know `banking_stage_scheduler_leader_detection`'s volume is quite low because it's only from leaders and per slot. how about commenting as this is intentional?

Reply:
I just reported same metrics as the old-scheduler leader metrics. I'm happy to remove this, as you said it's redundant.

Reply:
fe13db72464d51ef78c76bff9297978e1456b838